mne/simulation/evoked.py

```diff
@@ -106,10 +109,75 @@ def simulate_noise_evoked(evoked, cov, iir_filter=None, random_state=None):
     -----
     .. versionadded:: 0.10.0
     """
+    return _simulate_noise_evoked(evoked, cov, iir_filter, random_state)
+
+
+def _simulate_noise_evoked(evoked, cov, iir_filter, random_state):
     noise = evoked.copy()
-    noise.data = _generate_noise(evoked.info, cov, iir_filter, random_state,
-                                 evoked.data.shape[1])[0]
-    return noise
+    noise.data[:] = 0
+    return _add_noise(noise, cov, iir_filter, random_state,
+                      allow_subselection=False)
+
+
+@verbose
+def add_noise(inst, cov, iir_filter=None, random_state=None,
+              verbose=None):
+    """Create noise as a multivariate Gaussian.
+
+    The spatial covariance of the noise is given from the cov matrix.
+
+    Parameters
+    ----------
+    inst : instance of Evoked, Epochs, or Raw
```

Comment:
@larsoner I don't see any test for adding noise to evoked. Is the docstring wrong? or is test missing?

Reply:
No test for that, I can add it